### ISSUE TITLE
pushall emits a warning

### ISSUE THREAD

**Issue by mikedanese** [bazelbuild/rules_docker#94]
Using pushall to push a bundle always emits a warning:
```
docker/contrib/push-all.bzl:39:7: Pushing an image based on a tarball can be very expensive.  If the image is the output of a docker_build, consider dropping the '.tar' extension. If the image is checked in, consider using docker_import instead.
```
Any way around this?

**Comment by mattmoor** [comment]
Yeah, that means you are triggering a slow path.  Do you have a pointer to the push-all that I can walk to see if I can spot the problem?

**Comment by mikedanese** [comment]
I'm hitting it with pushs generated from this loop:

https://github.com/kubernetes/kubernetes/pull/49036/files#diff-47b6f163d969c2e3f9262ec436542cafR86

**Comment by mattmoor** [comment]
I think the problem is that you are forcing a tarball link with the `:image.tar` parts of the base entries [here](https://github.com/mikedanese/kubernetes/blob/96876dbd49eef55731c62dc6bec024ec8b85a79f/build/BUILD#L39).  Dropping those should make things faster and warning free.

**Comment by mattmoor** [comment]
@mikedanese Yeah, avoid `.tar` because it creates a less efficient form, which (due to limitations in skylark) we cannot invisibly pull apart.

Included in this is: don't even use `.tar` in `docker_push` if you can avoid it because we can actually upload things much more efficiently if we never enter tarball form.

**Comment by jschaf** [comment]
I keep hitting this warning but I'm not sure how I'm supposed to work around it. I'm building up a Docker image programmatically like so:

After some experimenting, the problem is using the tar output of `install_pkgs` as a container_image base. If I replace that with a normal image, it works.

Is there a way to "fix" the code or avoid the warning?

```
install_pkgs(
    name = "install_build_tool_pkgs",
)

pkg_tar(
    name = "entry_scripts_pkg",
)

container_layer(
  name = "node_binary_layer",
)

container_image(
    name = "runner_image",
    base = ":install_build_tool_pkgs.tar",
    tars = [
        ":entry_scripts_pkg",
    ],
    layers = [
        ":node_binary_layer",
    ],
    debs = ["@nodejs_yarn_deb//file"],
)
```
